graylog2-server/src/main/java/org/graylog2/indexer/counts/Counts.java

```diff
@@ -50,13 +57,22 @@ private long totalCount(final String[] indexNames) {
             return 0L;
         }
 
+        final Count request = new Count.Builder()
+                .query("{\"query\":{\"match_all\":{}}}")
+                .addIndex(Arrays.asList(indexNames))
+                .build();
+        final CountResult result;
         try {
-            return c.prepareSearch(indexNames)
-                    .setSize(0)
-                    .get()
-                    .getHits()
-                    .getTotalHits();
-        } catch (ElasticsearchException e) {
+            result = jestClient.execute(request);
+        } catch (IOException e) {
+            LOG.error("Couldn't fetch message count", e);
+            return -1L;
```

Comment:
Do we really want to return a C-style bogus value which we would need to interpret as an error at the consumer's level (and which would be just passed on as the actual value if we forget that) instead of throwing a higher-level exception at this place?